File: tests/ascii_only.pl
#!/usr/bin/perl
use warnings;
use strict;
use File::Find;

my $nonascii = 0;
sub process {
  /\.(c|cpp|h|i)$|^Makefile$/ or return;

  open(my $f, "<", $_) or die "Cannot open file $_: $!";
  while (<$f>) {
    if (/([^[:ascii:]])/) {
      $nonascii = 1;
      print "$File::Find::name: $_";
      last;
    }
  }
  close($f);
}
find(\&process, @ARGV ? $ARGV[0] : '.');
exit($nonascii);

File: tests/ascii_only.py
#!/usr/bin/env python3
import argparse
import glob
import os
import sys

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("root", default="", nargs="?", type=str, help="Root directory to check")
    args = parser.parse_args()

    nonascii = False
    for path in glob.iglob(os.path.join(args.root, "**"), recursive=True):
        if path.endswith((".c", ".cpp", ".h", ".i")) or "Makefile" in path:
            try:
                with open(path, "r", encoding="ascii") as path_file:
                    path_file.read()
            except:
                print("Non-ascii characters in {}".format(path), flush=True)
                nonascii = True

    sys.exit(1 if nonascii else 0)
